src/lib.rs

```diff
@@ -0,0 +1,154 @@
+use syn::{
+    export::{quote::quote, Span},
+    parse_macro_input, Ident, LitStr,
+};
+
+#[derive(Debug, Serialize, Deserialize)]
+struct WLanguage {
+    code: String,
+    name: String,
+    #[serde(skip_serializing)]
+    #[allow(dead_code)]
+    site: Vec<IgnoredAny>,
+    dir: String,
+    localname: String,
+    #[serde(skip)]
+    identifier: Option<Ident>,
+}
+
+#[derive(Debug, Deserialize)]
+struct SiteMatrix {
+    #[serde(skip_serializing)]
+    #[allow(dead_code)]
+    count: usize,
+    #[serde(flatten)]
+    languages: HashMap<String, WLanguage>,
+}
+#[derive(Debug, Deserialize)]
+struct SiteMatrixWrapper {
+    #[serde(rename = "sitematrix")]
+    site_matrix: SiteMatrix,
+}
+
+#[proc_macro]
+pub fn parse_languages(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as LitStr).value();
+    let file = File::open(input).unwrap();
+    let reader = BufReader::new(file);
+    let sitematrix: SiteMatrixWrapper = serde_json::from_reader(reader).unwrap();
+    let languages =
+        sitematrix
+            .site_matrix
+            .languages
+            .into_iter()
+            .fold(HashMap::new(), |mut acc, e| {
+                let mut e = e.1;
+                let name = if acc.contains_key(&e.localname) {
+                    normalize_string(&(e.localname.clone() + &e.code))
+                } else {
+                    normalize_string(&e.localname)
+                };
+                e.identifier = Some(Ident::new(&name, Span::call_site()));
+                acc.insert(name, e);
+                acc
+            });
+    let mut variants = quote!();
+    let mut language_data_arms = quote!();
+    let mut from_str_arms = quote!();
+    let mut array_def = quote!();
+    for (_key, value) in languages {
+        let ident = value.identifier.clone().unwrap();
+        let en_name = value.localname.clone();
+        let lang_name = value.name.clone();
+        let lang_code = value.code.clone();
+        variants = quote! {
+            #variants
+            #ident,
+        };
+        language_data_arms = quote! {
+            #language_data_arms
+            Language::#ident => (stringify!(#en_name), stringify!(#lang_name), stringify!(#lang_code)),
+        };
+        from_str_arms = quote! {
+            #from_str_arms
+            stringify!(#lang_code) | stringify!(#lang_name) => Language::#ident,
+        };
+        array_def = quote! {
+            #array_def
+            Language::#ident,
+        }
+    }
+
+    let expanded = quote! {
+
+        #[derive(Debug, Clone, Serialize)]
+        pub enum Language{
+            #variants
+        }
+
+        impl Language{
+            /// Returns the data associated to the language. It's formatted like this:
+            /// (Language name in English, Local Language name, Language Code)
+            fn language_data(&self) -> (&str, &str, &str) {
+                match self {
+                    #language_data_arms
```

Comment:
When viewing the languages, there are `"` around every language. Can they be removed? (if not it's alright, just an aesthetics thing

Reply:
Yeah I saw that yesterday and try to fix it don't know why it happens, will try again later